### ISSUE TITLE
Figure out how to incorporate workplace location into model flow

### ISSUE THREAD

**Issue by stefancoe** [psrc/urbansim2#122]
@hanase - I was able to run wplcm for all workers in the region, but had to reduce the sample size- it crashed using 100. If I understand correctly, in our current Urbansim implementation work at home & work location is run for the entire population as a pre-step before running the actual simulation. Then, these models are run yearly only for new persons and people changing jobs. 

So we could do a preliminary step and run for all workers before starting the simulation, but another idea I had is to get work location from Soundcast for the base year. The results from this model have been pretty well validated so it may be a good starting point. Since we are already getting skims from Soundcast, it would be pretty straightforward to get work location as well.  For now, i will try to run wplcm as a preliminary step. We also need to build a work at  home model, which I created a separate issue for. Thanks!

**Comment by msimonson-psrc** [comment]
That's an interesting idea Stefan - plunder WLCM results for Year 1 from SoundCast, port into UrbanSim, have it start from there...

**Comment by hanase** [comment]
It is an interesting idea. But we have some specifics built into the base year run to assure that the job totals do not change on a smaller geography. (A job that is assigned to a home-based worker gets assigned the location of the worker's residence, thus it moves.) What we do is the following:

1. In our pre-processing step (outside of urbansim), if a home-based job has been assigned to a residential building where there is a worker living, the two are connected. So all those workers are set as work-at-home.

2. For the remaining workers we run the work-at-home model (which also assigns home-based jobs to persons chosen to work at home), while it picks home-based jobs within the tract-city geography of the person's residence. 

3. For home-based jobs that are still not connected with workers by this time, the work-at-home model runs again, this time it tries to choose jobs within the large area of the worker's residence.

4. For workers that have chosen to work out of home, run the WLCM, using a choice set of size 30. 

The steps two and three assure that we don't have that much distortion in job counts in 2014. 

In SoundCast, you assign parcels to persons, right? It's not the individual jobs, is that right? That would complicate things, as we do need a direct connection between workers and jobs.

**Comment by hanase** [comment]
What we should do is to make the results of our worker-job models (that we run in the base year) a part of the base year cache. It has been on our list for some time, so this is a good excuse to do it. Then we don't have to run these models in every run. What do you think @pcaballero ?

**Comment by stefancoe** [comment]
Thanks Hana for the detailed description- that will help with implementation. And yeah, i think you are right about issues related to using Soundcast wplcm.

**Comment by stefancoe** [comment]
@hanase- do we currently use a rate to determine who changes job/workplace location in any given year. I'd like to implement something similar to the 'jobs_relocation' step. Thanks!
